packages/onboarding/src/step-container/index.tsx

```diff
@@ -0,0 +1,197 @@
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { ReactChild, ReactElement } from 'react';
+import ActionButtons from '../action-buttons';
+import StepNavigationLink from '../step-navigation-link';
+import './style.scss';
+
+interface Props {
+	stepContent: ReactElement;
+	hideBack?: boolean;
+	hideSkip?: boolean;
+	hideNext?: boolean;
+	formattedHeader?: ReactElement;
+	shouldHideNavButtons?: boolean;
+	skipButtonAlign?: 'top' | 'bottom';
+	skipHeadingText?: string;
+	hideFormattedHeader?: boolean;
+	headerImageUrl?: string;
+	backLabelText?: string | ReactChild;
+	skipLabelText?: string | ReactChild;
+	nextLabelText?: string | ReactChild;
+	className?: string;
+	// Displays an <hr> above the skip button and adds more white space
+	isLargeSkipLayout?: boolean;
+	isExternalBackUrl?: boolean;
+	headerButton?: ReactElement;
+	customizedActionButtons?: ReactElement;
+	isWideLayout?: boolean;
+	isFullLayout?: boolean;
+	isHorizontalLayout?: boolean;
+	goBack?: () => void;
+	goNext?: () => void;
+	flowName?: string;
+	intent?: string;
+	stepName?: string;
+	stepSectionName?: string;
+	recordTracksEvent: ( eventName: string, eventProperties: object ) => void;
+}
+
+const StepContainer: React.FC< Props > = ( {
+	stepContent,
+	hideBack,
+	hideSkip,
+	hideNext = true,
+	formattedHeader,
+	shouldHideNavButtons,
+	skipButtonAlign = 'bottom',
+	skipHeadingText,
+	headerImageUrl,
+	hideFormattedHeader,
+	backLabelText,
+	skipLabelText,
+	nextLabelText,
+	className,
+	isHorizontalLayout,
+	isFullLayout,
+	isWideLayout,
+	headerButton,
+	isExternalBackUrl,
+	isLargeSkipLayout,
+	customizedActionButtons,
+	goBack,
+	goNext,
+	flowName,
+	intent,
+	stepName,
+	stepSectionName,
+	recordTracksEvent,
+} ) => {
+	const translate = useTranslate();
+
+	const recordClick = ( direction: 'back' | 'forward', stepSectionName?: string ) => {
+		const tracksProps = {
+			flow: flowName,
+			step: stepName,
+			intent: intent,
+		};
+
+		// We don't need to track if we are in the sub-steps since it's not really going back a step
+		if ( direction === 'back' && ! stepSectionName ) {
+			recordTracksEvent( 'calypso_signup_previous_step_button_click', tracksProps );
+		}
```

Comment:
@escapemanuele Has this event been registered at all? I can't seem to find it in https://mc.a8c.com/tracks/events/

No need to go ahead and register if not. I intend to remove this whole section (so hoping it's actually a dead event / never made it to analyses)

Reply:
Hey @chriskmnds, https://mc.a8c.com/tracks/events/event.php?eventname=calypso_signup_previous_step_button_click here it is, not registered as you say!

Reply:
thanks. found it. it looks like registered or not still makes it to funnels 

The question I wanted to get to was about:

> // We don't need to track if we are in the sub-steps since it's not really going back a step

Can you tell me more? When would rendering `BackButton` (the only `direction === 'back'` form the looks of it) not entail going back a step? Is there an example I can look at where the button is rendered but not going back a step?

Reply:
(sorry for dragging you a couple of years back in history! 😬)

Reply:
Ah you only extracted these parts to onboarding package from elsewhere/StepWrapper. never mind then. barking up the wrong tree :-)